onedal/linear_model/linear_model.py

```diff
@@ -198,8 +161,43 @@ def __init__(
         super().__init__(fit_intercept=fit_intercept, copy_X=copy_X, algorithm=algorithm)
 
     def fit(self, X, y, queue=None):
-        return super()._fit(X, y, _backend.linear_model.regression, queue)
+        module = _backend.linear_model.regression
 
-    def predict(self, X, queue=None):
-        y = super()._predict(X, _backend.linear_model.regression, queue)
-        return y
+        policy = self._get_policy(queue, X, y)
+
+        X_loc, y_loc = X, y
+        if not isinstance(X, np.ndarray):
+            X_loc = np.asarray(X)
+
+        dtype = get_dtype(X_loc)
+        if dtype not in [np.float32, np.float64]:
+            dtype = np.float64
+            X_loc = X_loc.astype(dtype, copy=self.copy_X)
+
+        y_loc = np.asarray(y_loc).astype(dtype=dtype)
+
+        # Finiteness is checked in the sklearnex wrapper
+        X_loc, y_loc = _check_X_y(X_loc, y_loc, force_all_finite=False, accept_2d_y=True)
+
+        self.n_features_in_ = _num_features(X_loc, fallback_1d=True)
+
+        X_loc, y_loc = _convert_to_supported(policy, X_loc, y_loc)
+        params = self._get_onedal_params(get_dtype(X_loc))
+        X_table, y_table = to_table(X_loc, y_loc)
+
+        hparams = get_hyperparameters("linear_regression", "train")
+        if hparams is not None and not hparams.is_default:
+            result = module.train(policy, params, hparams.backend, X_table, y_table)
+        else:
+            result = module.train(policy, params, X_table, y_table)
+
+        self._onedal_model = result.model
+
+        coeff = from_table(result.model.packed_coefficients)
+        self.coef_, self.intercept_ = coeff[:, 1:], coeff[:, 0]
+
+        if self.coef_.shape[0] == 1 and y_loc.ndim == 1:
```

Comment:
I feel like there is a more elegant solution for this

Reply:
https://github.com/intel/scikit-learn-intelex/pull/1656#discussion_r1467600543